### ISSUE TITLE
COPY GRANTS option fails on new views

### ISSUE THREAD

**Issue by scastria** [Snowflake-Labs/terraform-provider-snowflake#2519]
### Terraform CLI and Provider Versions

TF v1.5.2
Snowflake TF 0.80.0

### Terraform Configuration

```terraform
resource "snowflake_view" "CustomerView" {
  provider = snowflake.READER_ACCOUNTADMIN
  for_each = local.customer_views
  database = snowflake_database.CustomerDatabase[split(".", each.key)[0]].name
  schema = snowflake_schema.CustomerSchema[join(".", [split(".", each.key)[0], split(".", each.key)[1]])].name
  name = split(".", each.key)[2]
  statement = <<-SQL
    -- Force a change to the view so that it is recompiled ${timestamp()}
    SELECT * FROM DB_SHARE.${split(".", each.key)[1]}.${split(".", each.key)[2]}
  SQL
  is_secure = true
  copy_grants = true
}
```

### Expected Behavior

View is created successfully

### Actual Behavior

verbose TF_LOG shows that the CREATE VIEW statement includes the COPY GRANTS clause since copy_grants = true above.  However, since this is a brand new view and NOT a CREATE OR REPLACE, COPY GRANTS is invalid and you get a SQL error of: "Invalid operation COPY GRANTS without specifying source object".  Therefore, if the Snowflake provider knows this is a brand new view, then it should omit COPY GRANTS even if copy_grants = true so that valid SQL is produced.

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

Medium

**Comment by sfc-gh-asawicki** [comment]
Also, I see you are using "." splitting in your configurations. Snowflake identifiers are a bit more complex; it's possible to have the dot character as part of the identifier if you have a fully qualified part (wrapped in double quotes). We have the identifiers rework incoming. We want to address the issues our users have with identifiers.

This is just a FYI because these identifiers can really hurt sometimes.

**Comment by sfc-gh-asawicki** [comment]
Hey @scastria . We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!

**Comment by sfc-gh-asawicki** [comment]
Closing due to inactivity. Please create a new ticket with the newest provider version if the problem is still encountered.
